Improvements to gui [UPDATED]

GuillaumeBrochu opened the pull request:
- correct English error in gui ("Finnish" vs ~~"Finished"~~ "Completed!")
- add gui and gui_lex.txt to distrib: in Makefile
- remove dead code (config_gwd_file, config_only_file)
- remove duplicated definition of "config.txt"
- add share_dir for -hd argument of gwd (needed for packaging in Debian)
- start gwd using the -lang argument
- <s>close_server now more robust to multiple gwd instances</s> (REMOVED, addressed separately in https://github.com/geneweb/geneweb/issues/547)
- indicate the location of the configuration file in the assistant
- shorten a line in the French translation of the assistant

GuillaumeBrochu committed:
Add gui and gui_lex.txt to distrib: in Makefile

GuillaumeBrochu commented:
N.B. : gui can be compiled using "make everything"

hgouraud reviewed (commented):
If "Terminer" is correct in French,  then it should be "Finish", otherwise  the French should be "Terminé"

GuillaumeBrochu commented:
Yes, of course. I will correct this. Thank you!

GuillaumeBrochu committed:
Correct English wording in gui

Finnish -> Completed!

Signed-off-by: GuillaumeBrochu <guillaume_brochu@yahoo.ca>

GuillaumeBrochu commented:
I also noticed that this test in "launch_server":
https://github.com/geneweb/geneweb/blob/master/gui/gui.ml#L1115-L1124
is not working to detect multiple gwd instances on Linux. <s>That's why I have modified "close_server" to avoid to stop another instance of gwd also listening on the same port but that was not launched by gui.</s>

There is also probably something to improve here, but I'll need your ideas.

fablhx commented:
Oh. I never though one would launch more than one instance. Is it a concrete use case ?

GuillaumeBrochu commented:
It happens during my tests on the geneweb packages for Debian.

GuillaumeBrochu commented:
Should I split this PR in two parts :  the additional check for the server launch in one PR, and the other improvements in a second one?

GuillaumeBrochu committed:
Improvements to gui:

- remove dead code (config_gwd_file, config_only_file)
- remove duplicated definition of "config.txt"
- factorize share_dir for -hd argument of gwd (needed for packaging in Debian)
- start gwd using the -lang argument
- indicate the location of the configuration file in the assistant
- shorten a line in the French translation of the assistant

GuillaumeBrochu commented:
UPDATED. I removed the change to close_server. Will be addressed later in a separate issue (==> https://github.com/geneweb/geneweb/issues/547).

EDIT : Tested on linux with "make everyting". Of course, as gwc2 is not perfectly working yet on Geneweb V7, one should use gwc to create a base manually (or with gwsetup) in order to test gui.

GuillaumeBrochu commented:
Git-rebased and validated with "make everything" by AppVeyor. What do you think? Should I improve something?

GuillaumeBrochu commented:
Please also review this one.

hgouraud commented:
No opinion! I dont play with GUI.
Allowing several gwd processes to run is a good question. Most of the time, you want to kill previous processes before starting a new one. We have seen recent situations where presence of a second gwd process confused the bug chasing subject!! In my case, killing other gwd processes is done in the launch script! Forcing it in gwd itself is possibly too restrictive. One could have two processes listening on different ports! But then, is the base.lck mechanism solid enough to guarantee that these two processes will not step on each other toes? For lack of definitive answer to this question, I would fall back on the single gwd process situation.

GuillaumeBrochu commented:
@hgouraud Few months ago, I removed the proposed modifications related to launch_server, close_server, and conflicts between gwd instances.  I opened a separate issue about that. They are no longer in this PR.

The remaining elements in this PR are only small improvements to help for the packaging and the use of gui. I have included these small improvements in a patch in Debian (otherwise gui was broken and unusable), but as these are general improvements (not Debian-specific), I think it's worth to push them upstream.
